### PR TITLE
[DM-27342] Fix some specification errors

### PR DESCRIPTION
**Pull request by rra**
Fix a few things that turned up during implementation:

- scopes may be NULL in SQL since using an empty string is weird
- Specify how to return a new token
- Add last_used to the token info
- Improve the index used for finding existing internal tokens

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -528,6 +528,13 @@ For all routes listed below with a ``username`` path parameter, only administrat
     Only user tokens may be created this way.
     Tokens of other types are created through non-API flows described later.
     The token name, scopes, and desired expiration are provided as parameters.
+    The newly-created token is returned as follows:
+
+    .. code-block:: json
+
+       {
+         "token": "gt-qVGZIh65TAJlNprOaMDhwg.WlUA5zyAY16dDRvDYxnwhg"
```
Comment (ktlim) [inline index.rst:549]: Is it `gt` or `gsh`?

**Review by ktlim — COMMENTED** [review]
A few other questions:

- Why not have `child` be the primary key in the `subtoken` table?  They should be unique, right?
- Since usernames are not validated, an admin could remove all valid users but leave an invalid one.  We don't need to prevent this foot-shooting, but maybe the UI should warn against it when deleting the current user from the `admin` table.
- Should we anonymize/mask IP addresses at some point?

File: index.rst (hunk 1 above)
Comment (rra) [reply index.rst:549]: `gt-`, sorry. Fixed in the next version.

**Comment by rra** [comment]
- Good call, yes, `child` can be the primary key. Made that fix.
- The admin handling is a temporary stopgap until we have the group component up and running. I've made that more explicit and added a note to the admins API.
- Good question on IP redaction. It makes them a lot less useful for analyzing problems, but it would also reduce the risk of the database being compromised somewhat, and possibly would help with GDPR for deployments that need that. I'm not sure what to do about that and probably need to do more reading on GDPR, but I'll add a note.

**Pull request merged**